File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.js
function solution(triangle) {
    const dp = triangle.slice()
    
    for (let i = dp.length-2; i >= 0; i--) {
        for (let j = 0; j < dp[i].length; j++) {
            dp[i][j] += Math.max(dp[i+1][j], dp[i+1][j+1])
        }
    }
    return dp[0][0]
}

File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.py
def solution(triangle):
    dp = [row[:] for row in triangle]
    
    for i in range(len(dp) - 2, -1, -1):
        for j in range(len(dp[i])):
            dp[i][j] += max(dp[i+1][j], dp[i+1][j+1])
    return dp[0][0]